pkg/workloads/console/controller.go

```diff
@@ -221,8 +222,12 @@ func (r *reconciler) Reconcile() (res reconcile.Result, err error) {
 	}
 
 	switch {
-	case r.console.Pending():
-		res = requeueAfterInterval(r.logger, time.Second)
+	case r.console.PreRunning():
+		if r.console.Pending() {
+			res = requeueAfterInterval(r.logger, time.Second)
+		} else {
+			res = requeueAfterInterval(r.logger, r.console.TTLSecondsBeforeRunning())
```

Comment:
This _might_ be technically wrong? (If so, it definitely was before too, but we could take this opportunity to fix it up)

I'm thinking this timeline:

Timestamp | Event
--------- | ------
0s | Create a console, with TTLSecondBeforeRunning of 300s
50s | Controller is restarted. A new watch on the object starts, and we hit this code
300s | We should have begun reconciling the console here, because we hit (CreationTimestamp + TTLSecondsBeforeRunning)
350s | We actually begin reconciling here. The console is GC'd 50s after it should have been.

Reply:
Ah yeah good catch!